spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/iceberg/IcebergPropertiesConverter.java

```diff
@@ -49,7 +49,21 @@ public Map<String, String> toSparkCatalogProperties(Map<String, String> properti
     Preconditions.checkArgument(
         StringUtils.isNotBlank(catalogBackend),
         String.format("%s should not be empty", IcebergConstants.CATALOG_BACKEND));
-    all.put(IcebergPropertiesConstants.ICEBERG_CATALOG_TYPE, catalogBackend);
+    if (catalogBackend.equalsIgnoreCase(
+        IcebergPropertiesConstants.ICEBERG_CATALOG_BACKEND_CUSTOM)) {
+      String catalogBackendImpl = all.remove(IcebergConstants.CATALOG_BACKEND_IMPL);
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(catalogBackendImpl),
+          String.format(
+              "%s should not be empty when %s is %s",
+              IcebergConstants.CATALOG_BACKEND_IMPL,
+              IcebergConstants.CATALOG_BACKEND,
+              IcebergPropertiesConstants.ICEBERG_CATALOG_BACKEND_CUSTOM));
+      all.put(IcebergPropertiesConstants.ICEBERG_CATALOG_IMPL, catalogBackendImpl);
```

Comment:
The `IcebergPropertiesConstants.ICEBERG_CATALOG_IMPL` is actually 
the `CatalogProperties.CATALOG_IMPL`. What do we gain from cloning this constants rather than referencing it directly? Is there a dependency circle we are trying to avoid?

Reply:
It's to keep the code format consistent,just like `IcebergPropertiesConstants.ICEBERG_CATALOG_TYPE`, it is also actually the `CatalogUtil.ICEBERG_CATALOG_TYPE`